### ISSUE TITLE
Error at ...video_player_windows/windows/CMakeLists.txt:11 (file):   file DOWNLOAD HASH mismatch

### ISSUE THREAD

**Issue by insinfo** [anirudhb/flutter_packages#3]
I'm getting this error when compiling the app

```console
PS C:\MyDartProjects\riodasostras\riodasostrasapp_mob> flutter run
Warning: You are using these overridden dependencies:
! rxdart 0.27.3
Running "flutter pub get" in riodasostrasapp_mob...              2.857ms
Launching lib\main.dart on Windows in debug mode...
CMake Error at flutter/ephemeral/.plugin_symlinks/video_player_windows/windows/CMakeLists.txt:11 (file):
  file DOWNLOAD HASH mismatch

    for file: [C:/MyDartProjects/riodasostras/riodasostrasapp_mob/build/windows/plugins/video_player_windows/ffmpeg_libs/ffmpeg-4.4-full_build-shared.7z]        
      expected hash: [e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855]
        actual hash: [146d0d5ab6ba50e6813bab47518bdc582abe122bb2d41bcc5609314890df460c]
             status: [0;"No error"]



CMake Error at C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find FFmpeg (missing: FFMPEG_LIBRARIES FFMPEG_INCLUDE_DIRS
  AVFORMAT_LIBRARIES AVFORMAT_INCLUDE_DIRS AVCODEC_LIBRARIES
  AVCODEC_INCLUDE_DIRS AVFILTER_LIBRARIES AVFILTER_INCLUDE_DIRS
  AVUTIL_LIBRARIES AVUTIL_INCLUDE_DIRS SWSCALE_LIBRARIES SWSCALE_INCLUDE_DIRS
  SWRESAMPLE_LIBRARIES SWRESAMPLE_INCLUDE_DIRS POSTPROC_LIBRARIES
  POSTPROC_INCLUDE_DIRS)
Call Stack (most recent call first):
  C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  flutter/ephemeral/.plugin_symlinks/video_player_windows/windows/FindFFmpeg.cmake:151 (find_package_handle_standard_args)
  flutter/ephemeral/.plugin_symlinks/video_player_windows/windows/CMakeLists.txt:20 (find_package)


Building Windows application...
Exception: Unable to generate build files
PS C:\MyDartProjects\riodasostras\riodasostrasapp_mob> 
```

```dart
import 'package:flutter/material.dart';
import 'package:riodasostrasapp/app/shared/theme/style.dart';
import 'package:video_player/video_player.dart';

class TurismoHomePage extends StatefulWidget {
  @override
  State<TurismoHomePage> createState() => _TurismoHomePageState();
}

class _TurismoHomePageState extends State<TurismoHomePage> {

  late VideoPlayerController _controller;
  @override
  void initState() {   
    super.initState();
    _controller = VideoPlayerController.network(
        'https://www.experimenteriodasostras.com.br/storage/turismo/midias/ffec8800-09d2-45a4-9b99-09ab04b057bf.mp4')
      ..initialize().then((_) {
        // Ensure the first frame is shown after the video is initialized, even before the play button has been pressed.
        setState(() {});
      });
  }

  @override
  Widget build(BuildContext context) {
    var screenSize = MediaQuery.of(context).size;

    return Scaffold(
      appBar: appBarDefault(title: 'Turismo'),
      body: Container(     
        child: SingleChildScrollView(
          child: Column(
            children: [
              //video
              Container(
                color: Colors.amber,
                width: screenSize.width,
                child: _controller.value.isInitialized
                    ? AspectRatio(
                        aspectRatio: _controller.value.aspectRatio,
                        child: VideoPlayer(_controller),
                      )
                    : Container(),              
              ),
              //menu grid
              Container(
                padding: EdgeInsets.all(16),
                width: screenSize.width,
                child: Wrap(
                  spacing: 16,
                  runSpacing: 16,
                  children: [
                    Container(
                      decoration: BoxDecoration(
                          color: Colors.white,
                          borderRadius: BorderRadius.all(Radius.circular(10))),
                      padding: EdgeInsets.all(10),
                      child: Column(
                        children: [
                          Text(
                            '',
                            style: TextStyle(
                                fontFamily: 'pictogramas_turismo',
                                fontSize: 71),
                          )
                        ],
                      ),
                    ),
                  ],
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  @override
  void dispose() {
    
    super.dispose();
    _controller.dispose();
  }
}

```

### pubspec.yaml

```yaml
 video_player: ^2.1.1
  video_player_windows:
    git:
      url: https://github.com/anirudhb/flutter_packages.git
      path: video_player_windows
```

```
PS C:\MyDartProjects\riodasostras\riodasostrasapp_mob> flutter doctor -v
[√] Flutter (Channel stable, 2.8.1, on Microsoft Windows [versÃ£o 10.0.22000.434], locale pt-BR)
    • Flutter version 2.8.1 at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (6 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\Users\isaque\AppData\Local\Android\sdk
    • Platform android-32, build-tools 32.0.0
    • Java binary at: C:\Program Files\Android\Android Studio1\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31702.278
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.63.2)
    • VS Code at C:\Users\isaque\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.32.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [versÃ£o 10.0.22000.434]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 97.0.4692.71
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 96.0.1054.62

• No issues found!
PS C:\MyDartProjects\riodasostras\riodasostrasapp_mob> 
```

**Comment by insinfo** [comment]
I commented the lines that download and extract the ffmpeg-4.4.1-full_build-shared.7z package, downloaded it manually and put a directory in the local dico c:\ and edited the line that defines the path to the ffmpeg directory and now it's compiling and running.

The only problem I'm having right now is that the video is flashing and doesn't loop.

```cmake
cmake_minimum_required(VERSION 3.15)
set(PROJECT_NAME "video_player_windows")
project(${PROJECT_NAME} LANGUAGES CXX)

include(FindFFmpeg.cmake)
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/")
#set(FFMPEG_HOME "${CMAKE_CURRENT_BINARY_DIR}/ffmpeg_libs/ffmpeg-4.4-full_build-shared")
set(FFMPEG_HOME "C:/src/ffmpeg-4.4.1-full_build-shared")
set(CMAKE_LIBRARY_PATH ${CMAKE_LIBRARY_PATH} "${FFMPEG_HOME}/lib")
set(CMAKE_INCLUDE_PATH ${CMAKE_INCLUDE_PATH} "${FFMPEG_HOME}/include")
 
#file(DOWNLOAD https://www.gyan.dev/ffmpeg/builds/packages/ffmpeg-4.4.1-full_build-shared.7z "${FFMPEG_HOME}.7z"
#  SHOW_PROGRESS
#  EXPECTED_HASH SHA256=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
#)
#file(ARCHIVE_EXTRACT INPUT "${FFMPEG_HOME}.7z"
#  DESTINATION "${FFMPEG_HOME}/.."
#  PATTERNS "ffmpeg-*/*"
#)

...
```
### log 
```
[INFO ] video_player_texture.cpp:244:VideoPlayerTexture::SetupEventChannel Registering channel with name flutter.io/videoPlayer/videoEvents1656984218000








[INFO ] video_player_texture.cpp:101:VideoPlayerTexture::Initialize Video codec name: h264
[INFO ] video_player_texture.cpp:107:VideoPlayerTexture::Initialize FPS: 29
[INFO ] video_player_stream_handler.cpp:17:VideoPlayerStreamHandler::ConstructInitialized 26893ms, 1920x1080
[INFO ] video_player_texture.cpp:260:VideoPlayerTexture::DecodeThreadProc Max queue items: 25
[INFO ] video_player_texture.cpp:261:VideoPlayerTexture::DecodeThreadProc Max audio queue items: 2377
[INFO ] video_player_windows_plugin.cpp:197:`anonymous-namespace'::VideoPlayerWindowsPlugin::HandleSetVolume Volume set to 1
flutter: initState true
[INFO ] video_player_texture.cpp:509:VideoPlayerTexture::ReadFrame Done with video
[INFO ] video_player_texture.cpp:550:VideoPlayerTexture::SendCompleted Sending completed event
[INFO ] video_player_texture.cpp:509:VideoPlayerTexture::ReadFrame Done with video

```
![Captura de Tela (47)](https://user-images.githubusercontent.com/12227024/150840969-08bfe9a8-9e15-46e0-b273-d5b211e7207d.png)
![Captura de Tela (46)](https://user-images.githubusercontent.com/12227024/150840977-85d0c800-cf40-4a9b-aff3-10c91d40a3ab.png)


https://user-images.githubusercontent.com/12227024/150841089-8ed71329-46fc-4273-a529-be92fd540f4b.mp4

**Comment by anirudhb** [comment]
Should be fixed in #2 now
